### ISSUE TITLE
Batch size checking is 32KB instead of 500KB, plus send.php flags requests as succesful even if they weren't even flushed.

### ISSUE THREAD

**Issue by MindfulPol** [segmentio/analytics-php#167]
### How to reproduce:
Setup:
1. Execute send.php file with an analytics.log file with track events smaller than 32kb but bigger than 500kb when grouped by 100 (100messages of 31kb will exceed 500kb and lib will fail instead of automatically reducing the batch to 500kb).

Outcome:
- Payload won't be flushed plus it'll tell you it has been sent successfully, also the requests(analytics.log) will be deleted losing all traceability of requests that haven't even been tried to be sent once eventhough they're smaller than 32kb and fulfill all other requirements to be a valid individual message. All of this was because the batch created by send.php exceeded 500kb. 

### Fix suggestions:

- Modify batch size length check to allow up to 500KB instead of 32KB and update endpoints from v1/import to v1/batch. (checked the python library to doublecheck and it indeed allows batches to be as big as 500KB)
- Modify send.php so it takes into account Segment::flush output to flag how many requests were successful and avoid creating batch requests bigger than 500KB

**Comment by MindfulPol** [comment]
Fixed this on my end by editing the file send.php itself, now if a request failed it won't be triggered as successful, plus instead of setting a lower flush_at limit I did some code so that the cron itself creates a batch as close as 32KB as possible depending on the events logged.

Maybe this could be refactored into a function inside the QueueConsumer/LibConsumer, what do you think? I needed a quick fix so that's why I didn't bother thinking about where it would be best to put it, since editing the send.php file itself seems a bit sloppy, let me know what do you think :D

Thank you!

**Comment by nd4p90x** [comment]
@MindfulPol Thank you, if I could ask a few questions before I dig too deep into this issue, what version of analytics-php are you currently using?  Did you modify any of the base options in your implementation?  Thanks

**Comment by MindfulPol** [comment]
Hey @nd4p90x,
Issue affects the latest version of the library, basically:
- old endpoints for http api
- wrong batch size check
- send.php flags requests as successful even if they failed
Because of this we decided to wrap the HTTP API ourselves, since it could lead to some errors where data would get lost. The pull request I provided would be a 'functional' version of the library which addresses the mentioned points. 
May I help you with anything just ask :). Thank you
